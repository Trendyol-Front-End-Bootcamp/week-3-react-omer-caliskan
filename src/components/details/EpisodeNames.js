import React, { useState } from 'react';
import { getEpisodeName } from "../../service/RickandMortyService"

function EpisodeNames( {episodeUrl, epNameforTest} ){
const [episodeName, setEpisodeName] = useState("");

const fetch = async () => {
        const data = await getEpisodeName(episodeUrl);
        setEpisodeName(data)
      }

      fetch();

return(
    <li>{epNameforTest || episodeName}</li>
)

}

export default EpisodeNames;