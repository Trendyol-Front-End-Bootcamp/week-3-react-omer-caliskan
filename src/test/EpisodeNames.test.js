import React from 'react'
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import EpisodeNames from "../components/details/EpisodeNames";

Enzyme.configure({ adapter: new Adapter() })

describe("episode names", () => {
    let episodeNames
    let episodeName
    beforeEach(async () => {
        episodeName = "Star Mort: Rickturn of the Jerri"
        episodeNames = shallow(<EpisodeNames epNameforTest={episodeName} />)
        
    })
    it("should render episode names", () => {

        expect(episodeNames.debug()).toMatchSnapshot()
    })

    it("should render episode names correctly", () => {
        let cardText = episodeNames.find('li').text()

        expect(episodeName).toEqual(cardText)
    })
})