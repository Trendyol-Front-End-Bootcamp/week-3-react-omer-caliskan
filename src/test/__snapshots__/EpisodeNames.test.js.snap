// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`episode names should render episode names 1`] = `
"<li>
  Star Mort: Rickturn of the Jerri
</li>"
`;
